db/helpers/collectorUtils.js

```diff
@@ -82,8 +82,26 @@ function validateVersion(version) {
   }
 }
 
+/**
+ * Find unassigned generators and assign them.
+ * @param {Object} seq - Sequelize object
+ * @returns {Promise} - Resolves to array of assigned generator db objects
+ */
+function findAndAssignGenerators(seq) {
+  return seq.models.Generator.findAll(
+    { where: { currentCollector: null, isActive: true } }
+  )
+  .then((unassignedGenerators) =>
+    Promise.all(unassignedGenerators.map((g) => {
+      g.assignToCollector();
```

Comment:
I feel like this assignToCollector (which calls seq "find") should return a promise which should be resolved before calling g.save ... otherwise do we risk saving g *before* a collector has actually been assigned?